### ISSUE TITLE
[mac] buildcurl.sh requires bash without saying so / broken on macOS 10.15

### ISSUE THREAD

**Issue by brevilo** [BOINC/boinc#3474]
**Describe the bug**
`buildcurl.sh` [requires](https://github.com/BOINC/boinc/blob/master/mac_build/buildcurl.sh#L150) `bash` it seems. That's a requirement which is neither explicitly stated AFAIK and which is also not a given on macOS 10.15 anymore as it (finally) uses `zsh` as default shell.

**Steps To Reproduce**
1. Run `buildcurl.sh` in `zsh`

**Expected behavior**
A working script

**System Information**
 - OS: macOS 10.15 (or just zsh)
 - BOINC Version: master

**Comment by CharlieFenton** [comment]
What issues have you found to lead you to believe that you can't still use bash? My understanding is that, even though zsh is now the _default_ shell under OS 10.15 , you can still use bash, and that bash is used if the script begins with `#!/bin/bash`. I also understand that zsh is mostly (though not completely) compatible with bash. Please see [this](https://apple.stackexchange.com/questions/361870/what-are-the-practical-differences-between-bash-and-zsh).

**Comment by brevilo** [comment]
> lead you to believe that you can't still use bash

No, I said the script fails in zsh, so one has to switch to bash manually to get it working again, which is what I did, of course. So as it is right now, a casual 10.15 user with its default shell will get a build failure. I described me expectations in "Expected behavior". IOW, the script itself should deal with that and be effectively shell-agnostic.

**Comment by brevilo** [comment]
> bash is used if the script begins with `#!/bin/bash`

You sure that's the case when you `source` a script (the documented BOINC
way) instead of executing it?

(sorry for the late update but GitHub is just broken today)

**Comment by CharlieFenton** [comment]
I booted into my OS 10.15.3 partition, and I can confirm that the `setupForBOINC.sh` script (which invokes `buildcurl.sh`) fails when I set _zsh_ as my standard shell. I had not seen this before because when I installed OS 10.15, I migrated my settings from an earlier version of OS X, so _bash_ remained my default / standard shell.

I also encountered a build error for _wxWidgets 3.1.0_, even when using _bash_, which I have not seen on earlier versions of OS X, apparently because the headers `HIWebView.h` and `CarbonUtils.h` have been removed from the WebKit framework in the _Mac OSX 10.15.sdk_ in _Xcode 11_ . So that is another issue (though only for building _BOINC Manager_, not for building the BOINC libraries.)

**Comment by brevilo** [comment]
> I also encountered a build error for wxWidgets 3.1.0

Just for completeness for others reading this: that's dealt with in #3476
